src/zenml/integrations/mlflow/experiment_trackers/mlflow_experiment_tracker.py

```diff
@@ -232,21 +236,53 @@ def disable_autologging(self) -> None:
             xgboost,
         )
 
+        try:
+            if framework == "tensorflow":
+                tensorflow.autolog(disable=True)
+            elif framework == "gluon":
+                gluon.autolog(disable=True)
+            elif framework == "xgboost":
+                xgboost.autolog(disable=True)
+            elif framework == "lightgbm":
+                lightgbm.autolog(disable=True)
+            elif framework == "statsmodels":
+                statsmodels.autolog(disable=True)
+            elif framework == "spark":
+                spark.autolog(disable=True)
+            elif framework == "sklearn":
+                sklearn.autolog(disable=True)
+            elif framework == "fastai":
+                fastai.autolog(disable=True)
+            elif framework == "pytorch":
+                pytorch.autolog(disable=True)
+            else:
+                raise ValueError(f"Unknown framework: {framework}")
+
+        except Exception as e:
+            logger.warning(
+                f"Failed to disable MLflow autologging for framework "
+                f"{framework}: {e}"
+            )
+
+    def disable_autologging(self) -> None:
+        """Disables MLflow autologging."""
         # There is no way to disable auto-logging for all frameworks at once.
         # If auto-logging is explicitly enabled for a framework by calling its
         # autolog() method, it cannot be disabled by calling
         # `mlflow.autolog(disable=True)`. Therefore, we need to disable
         # auto-logging for all frameworks explicitly.
-
-        tensorflow.autolog(disable=True)
-        gluon.autolog(disable=True)
-        xgboost.autolog(disable=True)
-        lightgbm.autolog(disable=True)
-        statsmodels.autolog(disable=True)
-        spark.autolog(disable=True)
-        sklearn.autolog(disable=True)
-        fastai.autolog(disable=True)
-        pytorch.autolog(disable=True)
+        # To ensure that possible missing underlying packages aren't causing issues, we
+        # capture the import errors and log them as warnings.
+
+        self.disable_autologging_for_framework("tensorflow")
+        self.disable_autologging_for_framework("gluon")
+        self.disable_autologging_for_framework("xgboost")
+        self.disable_autologging_for_framework("lightgbm")
+        self.disable_autologging_for_framework("statsmodels")
+        self.disable_autologging_for_framework("spark")
+        self.disable_autologging_for_framework("sklearn")
+        self.disable_autologging_for_framework("fastai")
+        self.disable_autologging_for_framework("pytorch")
```

Comment:
I think we might make it a bit cleaner by using importlib, and have all frameworks in a list that we can extend when needed, but overall i think it looks good 
```suggestion

    def disable_autologging(self) -> None:
        """Disables MLflow autologging for all supported frameworks."""
        frameworks = [
            "tensorflow",
            "gluon",
            "xgboost",
            "lightgbm",
            "statsmodels",
            "spark",
            "sklearn",
            "fastai",
            "pytorch",
        ]

        for framework in frameworks:
            try:
                # Correctly prefix the module name with 'mlflow.'
                module_name = f"mlflow.{framework}"
                # Dynamically import the module corresponding to the framework
                module = __import__(module_name, fromlist=['autolog'])
                # Call the autolog function with disable=True
                module.autolog(disable=True)
            except ImportError as e:
                logger.warning(f"Module {module_name} not found. Skipping autologging disable for {framework}.")
            except Exception as e:
                logger.warning(
                    f"Failed to disable MLflow autologging for framework "
                    f"{framework}: {e}"
                )
```